proplet/wasm.go

```diff
@@ -15,28 +20,34 @@ import (
 var resultsTopic = "channels/%s/messages/control/proplet/results"
 
 type Runtime interface {
-	StartApp(ctx context.Context, wasmBinary []byte, id, functionName string, args ...uint64) error
+	StartApp(ctx context.Context, wasmBinary []byte, cliArgs []string, id, functionName string, args ...uint64) error
 	StopApp(ctx context.Context, id string) error
 }
 
 type wazeroRuntime struct {
-	mutex     sync.Mutex
-	runtimes  map[string]wazero.Runtime
-	pubsub    mqtt.PubSub
-	channelID string
-	logger    *slog.Logger
+	mutex           sync.Mutex
+	runtimes        map[string]wazero.Runtime
+	pubsub          mqtt.PubSub
+	channelID       string
+	logger          *slog.Logger
+	hostWasmRuntime string
 }
 
-func NewWazeroRuntime(logger *slog.Logger, pubsub mqtt.PubSub, channelID string) Runtime {
+func NewWazeroRuntime(logger *slog.Logger, pubsub mqtt.PubSub, channelID, hostWasmRuntime string) Runtime {
 	return &wazeroRuntime{
-		runtimes:  make(map[string]wazero.Runtime),
-		pubsub:    pubsub,
-		channelID: channelID,
-		logger:    logger,
+		runtimes:        make(map[string]wazero.Runtime),
+		pubsub:          pubsub,
+		channelID:       channelID,
+		logger:          logger,
+		hostWasmRuntime: hostWasmRuntime,
 	}
 }
 
-func (w *wazeroRuntime) StartApp(ctx context.Context, wasmBinary []byte, id, functionName string, args ...uint64) error {
+func (w *wazeroRuntime) StartApp(ctx context.Context, wasmBinary []byte, cliArgs []string, id, functionName string, args ...uint64) error {
+	if w.hostWasmRuntime != "" {
```

Comment:
This might be OK, but rather declare interface `WasmRuntime`, and then implement different `StartApp` functions for different runtimes (Wazero or Wasmtime or whatever)